src/actions/assetsActions.js

```diff
@@ -461,13 +467,16 @@ export const fetchAllAccountsTotalBalancesAction = () => {
     }
 
     const currency = fiatCurrencySelector(getState());
-    const zapperUSDBasedRates = await getZapperFiatRates();
-    if (currency !== USD && !zapperUSDBasedRates) {
+
+    const zapperUsdBasedRates = await getZapperFiatRates();
+    if (!zapperUsdBasedRates) {
       dispatch({ type: SET_FETCHING_TOTAL_BALANCES, payload: false });
       reportErrorLog('fetchAllAccountsTotalBalancesAction failed: no zapperUSDBasedRates', { accountsAddresses });
       return;
     }
 
+    const usdRate = zapperUsdBasedRates[currency.toUpperCase()] ?? 1;
```

Comment:
Defensive programming:
This check is somewhat unsafe, as it assumes that if currency is not found, then it must be USD, Could we check explicitly for USD and then return 1, so that all unexpected currencies would return undefined?

Reply:
Thanks, this is also called late night programming :) I thought of not unblocking user from getting balances if rates are not returned, but in fact rate check is done above and actually if there's no rate we shouldn't update at all rather than show wrong balances, will update accordingly, hope it makes sense 👌